UniWatch/DataAccess/UserManager.cs

```diff
@@ -92,16 +94,15 @@ public Teacher GetTeacherById(int teacherId)
         }
 
         /// <summary>
-        /// Returns the user associated with the given identity
+        /// Returns the logged in user
         /// </summary>
-        /// <param name="identity">The id underlying identity object for the user</param>
-        /// <returns>The User associated with the given identity</returns>
-        public User GetUser(string identityId)
+        public User GetUser()
         {
-            var teacher = _db.Teachers.Where(t => t.IdentityId == identityId).FirstOrDefault();
+            var userId = HttpContext.Current.User.Identity.GetUserId<string>();
+            var teacher = _db.Teachers.Where(t => t.IdentityId == userId).FirstOrDefault();
 
             if(teacher == null)
-                return _db.Students.Where(s => s.IdentityId == identityId).FirstOrDefault();
+                return _db.Students.Where(s => s.IdentityId == userId).FirstOrDefault();
```

Comment:
Consider throwing an exception if teacher and student are both null. Maybe rewrite like this for extensibility:

```
...
var teacher = ...;
if(teacher != null)
  return teacher;

var student = ...;
if(student != null)
  return student

return null;
```